### PR TITLE
Bug fix for github issue number 34, the same as old trac system's ticket 111

### PR DESCRIPTION
**Pull request by nmemarsa**
number 111. Modified core/LIS_fileIOMode.F90, routine LIS_create_output_directory
Resolves: #34

### CODE REVIEW

**Comment by jvgeiger** [comment]
Thanks for the update.  You fixed the bug, but after further review, I realize that the feature itself no longer fits in.  Would you please remove the optional argument dir_name from the LIS_create_output_directory routine, including

```
!   \item [dir\_name]
!     name of the directory to override the above format
```
in the DESCRIPTION section?

Then would you please add a couple more commits to this pull request?

* First please reindent lines 268 through 282 to be consistent with the rest of the subroutine.
* And then please remove trailing whitespace characters?

**Comment by nmemarsa** [comment]
Hi Jim,

I applied your comments and added a new commit to my fix/34 branch. Please consider accepting my pull request, or let me know if I should initiate another pull request for the latest commit to be merged.

Also, the x11 issue I was facing was due to me having a new Mac. I had to install XQuartz and all is good now.

Thanks,
Nargess


Nargess Memarsadeghi, Ph.D.
NASA GSFC, Code 586
Tel:  301-286-2938
"Opinions are mine and not that of NASA"

File: lis/core/LIS_fileIOMod.F90 [hunk 1]
```diff
@@ -228,114 +227,45 @@ subroutine LIS_create_output_directory(mname,dir_name)
...
-         if (ios .ne. 0) then
-            write(LIS_logunit,*)'ERR creating directory ',trim(out_dname)
-            flush(LIS_logunit)
-         end if
-
-      endif      
    endif
 
- end subroutine LIS_create_output_directory
+   #if ( defined AIX )
```
Comment (jvgeiger) [inline lis/core/LIS_fileIOMod.F90]:
```suggestion
#if ( defined AIX )
```

File: lis/core/LIS_fileIOMod.F90 [hunk 2]
```diff
@@ -228,114 +227,45 @@ subroutine LIS_create_output_directory(mname,dir_name)
...
+      call system('mkdir -p '//trim(out_dname))
```
Comment (jvgeiger) [inline lis/core/LIS_fileIOMod.F90]:
```suggestion
   call system('mkdir -p '//trim(out_dname))
```

File: lis/core/LIS_fileIOMod.F90 [hunk 3]
```diff
@@ -228,114 +227,45 @@ subroutine LIS_create_output_directory(mname,dir_name)
...
+   #else
```
Comment (jvgeiger) [inline lis/core/LIS_fileIOMod.F90]:
```suggestion
#else
```

File: lis/core/LIS_fileIOMod.F90 [hunk 4]
```diff
@@ -228,114 +227,45 @@ subroutine LIS_create_output_directory(mname,dir_name)
...
+   ! EMK...Calls to 'system' fail when using SGI MPT as the MPI implementation
+   ! on Pleiades. We replace with a C wrapper function that calls the 'mkdir' 
+   ! standard POSIX function. 
+   !         ios = system('mkdir -p '//trim(out_dname))
+   c_string = trim(out_dname)
+   ios = LIS_create_subdirs(len_trim(c_string),trim(c_string))
+   #endif
```
Comment (jvgeiger) [inline lis/core/LIS_fileIOMod.F90]:
```suggestion
#endif
```

**Pull request merged**